larsim/EventGenerator/SingleGen_module.cc

```diff
@@ -490,6 +500,16 @@ namespace evgen{
     }
 
     produces< std::vector<simb::MCTruth> >();
+    produces< sumdata::RunData, art::InRun >();
+
+  }
+
+
+  //____________________________________________________________________________
+  void SingleGen::beginRun(art::Run& run)
+  {
+    art::ServiceHandle<geo::Geometry const> geo;
+    run.put(std::make_unique<sumdata::RunData>(geo->DetectorName()));
   }
```

Comment:
I suggest that if this product is intended to represent an entire run, then you be explicit about the range-of-validity:

```.cpp
run.put(std::make_unique<RunData>(...), art::fullRun());
```

It is then quite explicit what is going on.  With a future version of *art*, I hope to make the second argument required.

Reply:
In my interpretation it is not clear whether `RunData` is conceptually intended to be full run or run "fragment", but the _current_ content is definitely run-wide. In fact, it should be job-wide, but `art::Results` are cast out of standard modules.
I have implemented the suggestion.